scripts/Muon/GUI/ElementalAnalysis/Plotting/plotting_presenter.py

```diff
@@ -48,3 +59,89 @@ def add_moveable_vline(self, plot_name, x_value, y_minx, y_max, **kwargs):
 
     def add_moveable_hline(self, plot_name, y_value, x_min, x_max, **kwargs):
         pass
+
+    def removeSubplotConnection(self, slot):
+        self.view.subplotRemovedSignal.connect(slot)
+
+    def close(self):
+        if self.selectorWindow is not None:
+            self.closeSelectorWindow()
+        if self.rmWindow is not None:
+            self.closeRmWindow()
+
+    def add(self):
+        print("to do")
+
+    def rm(self):
+        names = self.view.subplot_names
+        # if the remove window is not visable
+        if self.rmWindow is not None:
+            self.raiseRmWindow()
+        # if the selector is not visable
+        elif self.selectorWindow is not None:
+            self.raiseSelectorWindow()
+        # if only one subplot just skip selector
+        elif len(names) == 1:
+            self.getRmWindow(names[0])
+        # if no selector and no remove window -> let user pick which subplot to
+        # change
+        else:
+            self.selectorWindow = self.createSelectWindow(names)
+            self.selectorWindow.subplotSelectorSignal.connect(self.getRmWindow)
+            self.selectorWindow.closeEventSignal.connect(
+                self.closeSelectorWindow)
+            self.selectorWindow.setMinimumSize(300, 100)
+            self.selectorWindow.show()
+
+    def createSelectWindow(self, names):
+        return SelectSubplot(names)
+
+    def raiseRmWindow(self):
+        self.rmWindow.raise_()
+
+    def raiseSelectorWindow(self):
+        self.selectorWindow.raise_()
+
+    def closeSelectorWindow(self):
+        if self.selectorWindow is not None:
+            self.selectorWindow.close
```

Comment:
I'm completely ignorant what is going on here, but just to avoid a possible problem: should there be `()` after `close`?

Reply:
It does not work if you put `()` after close. It is a property and therefore does not have brackets.